### DISCUSSION TITLE
Status from Supervisor

### DISCUSSION THREAD

**Discussion by janitza-luwe** [troglobit/watchdogd#37] [Q&A] [answered]
Hi together, 
is it possible to log some information for a failed supervisor watchdog?
Iam monitoring a app on a embedded device and it would be _very_ useful to get some Information about the failure reason (is process still running, stack-trace, memory-useage, current system load, ...).
In my case it would be enough to run a shell-script when Watchdog detects supervisor watchdog timeout. I could write a script that monitors the syslog - but that would be very ugly.

Any ideas about that? Does watchdogd currently have mechanisms in place to accomplish something like this?

**Comment by troglobit** [comment, marked as answer]
The manual page, in the source tree [`man/watchdogd.conf.5`](https://man.troglobit.com/man5/watchdogd.conf.5.html) has some information on this.  Here's an excerpt.

> **script** = `/path/to/script.sh`
>  When a supervised process fails to meet its deadline the supervisor by default performs an unconditional reset,
>  saving the reset cause first.  However, if a script is provided in this section it will be called instead:
>
>            script.sh supervisor CAUSE PID LABEL
>
>  The CAUSE value is documented in watchdogctl(1).
>
>  The LABEL can be any free form string the supervised process used when registering with the supervisor, hence it
>  is given as the last argument to the script.
>
>  The return value of the script determines how the system continues to operate: POSIX OK (0) means the script has
>  handled the situation in some manner and watchdogd stops supervising the offending process, a non-zero return
>  value from script means the script has either failed to handle the situation or prefers to delegate to watchdogd
>  to save the reset cause and perform the actual system reset.
```

This should probably also be mentioned in the top-level README.
